pkg/storage/unified/apistore/store.go

```diff
@@ -449,18 +449,20 @@ func (s *Storage) GuaranteedUpdate(
 			if err != nil {
 				return err
 			}
-			mmm, err := utils.MetaAccessor(existingObj)
-			if err != nil {
-				return err
-			}
-			mmm.SetResourceVersionInt64(rsp.ResourceVersion)
 
 			if err := preconditions.Check(key, existingObj); err != nil {
 				if attempt >= MaxUpdateAttempts {
 					return fmt.Errorf("precondition failed: %w", err)
 				}
 				continue
 			}
+
+			acc, err := utils.MetaAccessor(existingObj)
+			if err != nil {
+				return err
+			}
+			acc.SetResourceVersionInt64(rsp.ResourceVersion)
+			res.ResourceVersion = uint64(rsp.ResourceVersion)
```

Comment:
2 questions
1. I am struggling to understand line 465 🤔 . What is this trying to achieve
2. What's the rational for moving setting the RV after performing the precondition check.

Reply:
1. I don't know 😅 The preconditions fn checks for some [kubernetes related preconditions ](https://github.com/kubernetes/apiserver/blob/master/pkg/storage/interfaces.go#L136). Some of them, from what I understand (the name for example) have a retry mechanism to avoid name clashes. This makes me suspect we have several check attempts for the same reason.
2. none, really. I had removed it at some point thinking we wouldn't need the check and I just ended up adding it before k8s checks afterwards. I can leave it as it was :)